src/snowflake/snowpark/dataframe.py

```diff
@@ -2929,6 +2930,58 @@ def create_or_replace_view(
             ),
         )
 
+    @df_collect_api_telemetry
+    def create_or_replace_dynamic_table(
+        self,
+        name: Union[str, Iterable[str]],
+        *,
+        warehouse: str,
```

Comment:
can warehouse be optional in the SQL of creating dynamic table? Because we can get the current warehouse at runtime and add it for users

Reply:
Warehouses are required in the SQL for creating dynamic tables.  I think we'd like force users to specify the warehouse, especially since the warehouse will be used in the background for refreshing the DT